Remora.Rest/Json/StringEnumConverter.cs

```diff
@@ -117,4 +117,23 @@ public override void Write(Utf8JsonWriter writer, TEnum value, JsonSerializerOpt
 
         writer.WriteStringValue(_enumsToNames[value]);
     }
+
+    /// <inheritdoc />
+    public override TEnum ReadAsPropertyName(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```

Comment:
This is unlikely to work since `Read` looks for a JsonTokenType of String, but PropertyName is its own TokenType.

Reply:
Hmm. Did this because it's what STJ does, but I can look this over again

Reply:
Add unit test to verify :P